sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -1023,7 +1023,8 @@ def _handle_temp_and_staging_locations(self, validator):
   def validate(self, validator):
     errors = []
     if validator.is_service_runner():
-      errors.extend(self._handle_temp_and_staging_locations(validator))
+      if not self.view_as(TestOptions).dry_run:
```

Comment:
Let's add a comment stating "this needs Internet connection and/or credentials"

Reply:
Done.